sympy/stats/stochastic_process_types.py

```diff
@@ -617,3 +651,153 @@ def expectation(self, expr, condition=None, evaluate=True, **kwargs):
 
         raise NotImplementedError("Mechanism for handling (%s, %s) queries hasn't been "
                                 "implemented yet."%(expr, condition))
+
+class GeneratorMatrixOf(TransitionMatrixOf):
+    """
+    Assumes that the matrix is the generator matrix
+    of the process.
+    """
+
+    def __new__(cls, process, matrix):
+        if not isinstance(process, ContinuousMarkovChain):
+            raise ValueError("Currently only ContinuousMarkovChain "
+                                "support GeneratorMatrixOf.")
+        matrix = _matrix_checks(matrix)
+        return Basic.__new__(cls, process, matrix)
+
+class ContinuousMarkovChain(ContinuousTimeStochasticProcess, StochasticProcessUtil):
+    """
+    Represents continuous Markov chain.
```

Comment:
This class represents a continuous time, discrete state Markov process. To me it seems unusual to call this a "chain" when it is in continuous time. It also does not represent *any* continuous time Markov "chain" since it needs to have discrete state (unlike e.g. a Weiner process).

I dislike the terminology but more importantly whatever terminology you use you should provide a mathematical definition so that it is unambiguous to users.

Reply:
Sure, I will add the definitions along with the references.